lib/async_with/runner.ex

```diff
@@ -82,7 +82,8 @@ defmodule AsyncWith.Runner do
   A `timeout`, in milliseconds, must be provided to specify the maximum time
   allowed for this operation to complete.
   """
-  @spec run_nolink([map], non_neg_integer) :: {:ok, any} | {:error, any} | {:exit, any}
+  @spec run_nolink([map], non_neg_integer) ::
+          {:ok, any} | {:error | :nomatch | :norescue | :nocatch | :exit, any}
```

Comment:
Good catch!

I believe `:exit` errors are wrapped in `{:error, error}`:

```suggestion
          {:ok, any} | {:error | :nomatch | :norescue | :nocatch, any}
```

Reply:
Yes, they are. There is also no clause for `{:exit, any}` where `Runner.no_link/2` is called, so this is the only place that needs to be changed.